src/battle_util.c

```diff
@@ -1220,7 +1221,7 @@ u8 DoFieldEndTurnEffects(void)
                     }
                 }
                 gBattleStruct->turnSideTracker++;
-                if (effect)
+                if (effect != 0)
                     break;
             }
             if (!effect)
```

Comment:
I don't mind `if (effect)` here personally, but if you want explicit `!= 0` then these `!` should likewise be `== 0`